tools/main.cpp

```diff
@@ -378,7 +378,15 @@ int main(int argc, char **argv)
 
             case '?':
                 display_usage();
-                exit(0);
```

Comment:
I think what you need to do is check the value of `optopt`. It should be something else than 0 when there is an invalid option. See https://linux.die.net/man/3/getopt_long

Reply:
Any option that is not handled in a switch case would be directed to `default`, wouldn't it?
Does it make sense to put the logic there instead to check inside `?`? The default behaviour could be to `display_usage` and then exit with 1? Or maybe I do not fully understand the goal of this check?

Reply:
> Any option that is not handled in a switch case would be directed to default, wouldn't it?

No, get_opts / get_longopts returns `?` for any invalid options found. It does set `optopt` in such cases, so yes, when need to call `display_usage` and then return 0 or 1 depending on the value of 'optopt`

Reply:
I see. Thanks for the explanation.

Reply:
In the end the best would be to change the short option for `--help` to `-h`? I think using `-?` was not a very good idea. Then we can `exit(1)` on `?`.

Reply:
If it's O.K. to repurpose `-h` (formerly used as shorthand for `--page-height`, but now deprecated), I'm happy to do that.

Reply:
Great. Don't forget to change the travis script accordingly
https://github.com/rism-ch/verovio/blob/988636d2c31133f8253cd08c2877b151fe744023/travis/ci_precache.sh#L67